src/main/java/org/elasticsearch/index/aliases/IndexAliasesService.java

```diff
@@ -119,6 +126,66 @@ public Filter aliasFilter(String... aliases) {
         }
     }
 
+    public Set<FieldMapper<?>> aliasFields(String... aliases) {
+        if (aliases == null || aliases.length == 0) {
```

Comment:
wouldn't this much easier to read without all the special casing for length? I really think this should be ok to just go over everything all the time?